articles/150_ros_command_line.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: default
+title: ROS Command Line
+permalink: articles/ros_command_line.html
+abstract:
+  This article describes ROS 2 nodes command line capabilities and its syntax.
+author: '[Michel Hidalgo](https://github.com/hidmic)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
```

Comment:
Maybe add a short section about the implementation. Basically where the extraction from `argc` / `argv` happens and that all remaining arguments are still available for the user code to process.

Reply:
That makes sense. It also raises an interesting question. Assuming extraction happens in `rcl`, should we support arguments specific to upper layers e.g. `rclcpp` or `rclpy`? Those would have to be available unparsed, separate from user defined arguments.

Reply:
We could, but they could live in between the `--ros-args` and `--`, and `rcl` could just ignore them and leave them in a "left over args" output. Then `rclcpp` and/or `rclpy` can use those arguments or make warnings/errors because there are unknown parameters left over.